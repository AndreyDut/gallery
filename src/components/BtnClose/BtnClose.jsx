import React from "react";

const BtnClose = ({ onClose }) => {
  return (
    <button
      className="btn-close"
      onClick={onClose}
      style={{
        border: "none",
        background: "transparent",
        outline: "none",
        cursor: "pointer",
      }}
    >
      <svg
        width="34px"
        height="34px"
        viewBox="0 0 24 24"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g id="SVGRepo_bgCarrier" strokeWidth="0"></g>
        <g
          id="SVGRepo_tracerCarrier"
          strokeLinecap="round"
          strokeLinejoin="round"
        ></g>
        <g id="SVGRepo_iconCarrier">
          {" "}
          <g id="Menu / Close_LG">
            {" "}
            <path
              id="Vector"
              d="M21 21L12 12M12 12L3 3M12 12L21.0001 3M12 12L3 21.0001"
              stroke="#000000"
              strokeWidth="2"
              strokeLinecap="round"
              strokeLinejoin="round"
            ></path>{" "}
          </g>{" "}
        </g>
      </svg>
    </button>
  );
};

export default BtnClose;
