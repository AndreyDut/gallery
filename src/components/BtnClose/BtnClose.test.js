import React from 'react';
import renderer, { act } from 'react-test-renderer';
import BtnClose from './BtnClose';

test('работа BtnClose', () => {

  const component = renderer.create(
    <BtnClose onClose={() => console.log("close")}/>
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();


  const buttonElem = component.root.find( el => el.type=='button' ); 
  // и "нажмём" на неё

  buttonElem.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // "нажмём" кнопку ещё раз
  buttonElem.props.onClick();
  
  // и получаем окончательный снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
  
});
