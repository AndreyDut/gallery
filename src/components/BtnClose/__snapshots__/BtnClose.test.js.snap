// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа BtnClose 1`] = `
<button
  className="btn-close"
  onClick={[Function]}
  style={
    Object {
      "background": "transparent",
      "border": "none",
      "cursor": "pointer",
      "outline": "none",
    }
  }
>
  <svg
    fill="none"
    height="34px"
    viewBox="0 0 24 24"
    width="34px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      id="SVGRepo_bgCarrier"
      strokeWidth="0"
    />
    <g
      id="SVGRepo_tracerCarrier"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <g
      id="SVGRepo_iconCarrier"
    >
       
      <g
        id="Menu / Close_LG"
      >
         
        <path
          d="M21 21L12 12M12 12L3 3M12 12L21.0001 3M12 12L3 21.0001"
          id="Vector"
          stroke="#000000"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
         
      </g>
       
    </g>
  </svg>
</button>
`;

exports[`работа BtnClose 2`] = `
<button
  className="btn-close"
  onClick={[Function]}
  style={
    Object {
      "background": "transparent",
      "border": "none",
      "cursor": "pointer",
      "outline": "none",
    }
  }
>
  <svg
    fill="none"
    height="34px"
    viewBox="0 0 24 24"
    width="34px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      id="SVGRepo_bgCarrier"
      strokeWidth="0"
    />
    <g
      id="SVGRepo_tracerCarrier"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <g
      id="SVGRepo_iconCarrier"
    >
       
      <g
        id="Menu / Close_LG"
      >
         
        <path
          d="M21 21L12 12M12 12L3 3M12 12L21.0001 3M12 12L3 21.0001"
          id="Vector"
          stroke="#000000"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
         
      </g>
       
    </g>
  </svg>
</button>
`;

exports[`работа BtnClose 3`] = `
<button
  className="btn-close"
  onClick={[Function]}
  style={
    Object {
      "background": "transparent",
      "border": "none",
      "cursor": "pointer",
      "outline": "none",
    }
  }
>
  <svg
    fill="none"
    height="34px"
    viewBox="0 0 24 24"
    width="34px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      id="SVGRepo_bgCarrier"
      strokeWidth="0"
    />
    <g
      id="SVGRepo_tracerCarrier"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <g
      id="SVGRepo_iconCarrier"
    >
       
      <g
        id="Menu / Close_LG"
      >
         
        <path
          d="M21 21L12 12M12 12L3 3M12 12L21.0001 3M12 12L3 21.0001"
          id="Vector"
          stroke="#000000"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
         
      </g>
       
    </g>
  </svg>
</button>
`;
